### PR TITLE
Set up the active project store

### PR DESCRIPTION
**Pull request by atjn**
The active project is just an instance of the class `Project` that contains everything we need to display and interact with the automata being created.

There is a store `project` that allows for direct access to this `Project` class, but if everything used it, then any minor update to the project would require a full update to the entire UI.
Therefore we have a scoped store for each project member, which will not update when other scoped stores update. We also have some custom stores.

The store scope looks like:

 - project
   - name
   - components
      - activeView
   - systems
      - activeView
   - queries
   - systemDeclarations
   - globalDeclarations

An update to a store will trigger updates in all parent stores and child stores, but will not trigger updates in any sibling stores.

### CODE REVIEW

File: src/lib/classes/fileAdapter/FileAdapterTauri.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	Queries,
+	Declaration,
+	type Named,
+	type Raw,
+} from "$lib/classes/automaton";
+import { Project } from "$lib/classes/project/Project";
+import type { IFileAdapterImplementation } from "./FileAdapterImplementation";
+export class FileAdapterTauri implements IFileAdapterImplementation {
+	supported() {
+		return Boolean("__TAURI__" in globalThis);
```
Comment (KamyaPA) [inline src/lib/classes/fileAdapter/FileAdapterTauri.ts:22]: Jeg har sat det op i $lib/tauri.ts

File: src/lib/globalState/activeProject.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+let projectStore: Project | undefined;
+
+const projectSubscribers = new Set<Subscriber<Project | undefined>>();
+const memberSubscribers: {
+	[key in keyof Project]: Set<Subscriber<Project[key] | undefined>>;
+} = {
+	name: new Set(),
+	components: new Set(),
+	systems: new Set(),
```
Comment (KamyaPA) [inline src/lib/globalState/activeProject.ts:36]: Skal man kunne "subscribe" til individuelle "components" og "sets"

File: src/lib/globalState/activeProject.ts [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+		return unsubscribe;
+	}
+}
+
+export const name = new ProjectMember("name");
+export const components = new ProjectMember("components");
+export const systems = new ProjectMember("systems");
+export const queries = new ProjectMember("queries");
+export const systemDeclarations = new ProjectMember("systemDeclarations");
+export const globalDeclarations = new ProjectMember("globalDeclarations");
```
Comment (KamyaPA) [inline src/lib/globalState/activeProject.ts:127]: En lille kommentar der fortæller hvad der bliver exporteret

File: src/lib/globalState/activeProject.ts (hunk 2 above)
Comment (atjn) [reply src/lib/globalState/activeProject.ts:36]: No, this is only for internal use to keep track of the subscribers that subscribe to the members of the `Project`

Comment (atjn) [reply src/lib/globalState/activeProject.ts:36]: You cannot subscribe to an individual component, except in the case where you are subscribing to the active view. I don't see any other case where a specific subscribe would be necessary.

**Review by KamyaPA — COMMENTED** [review]
Jeg har tilføget nogle kommentare til hvor jeg måske tænker at der er lidt

File: src/lib/globalState/activeProject.ts (hunk 2 above)
Comment (atjn) [reply src/lib/globalState/activeProject.ts:36]: I assume this is resolved, otherwise let me know

**Review by NybyDK — APPROVED** [review]
this is looking pretty good my guy 👁👄👁

**Review by Jesper276 — APPROVED** [review]
LGTM DKWIGO!

**Review by KamyaPA — APPROVED** [review]
LGTM

**Pull request merged**